### PR TITLE
Specify scheme in the jolokia URL

### PR DESCRIPTION
**Pull request by gashcrumb**
Will require this PR -> https://github.com/kubernetes/kubernetes/pull/14876

@liggitt @rhuss FYI...

### CODE REVIEW

**Comment by rhuss** [comment]
thx ;-)

**Comment by liggitt** [comment]
Pity, I had a day set aside to work on this

**Comment by gashcrumb** [comment]
@liggitt oh, sorry.  I can take it back if you like :-)

File: assets/app/scripts/extensions/javaLink.js [hunk 1]
```diff
@@ -43,7 +43,8 @@ angular.module('openshiftConsoleExtensions', ['openshiftConsole'])
       if (!pod || !pod.status || pod.status.phase !== 'Running') {
         return;
       }
-      var podName = pod.metadata.name;
+      // always use https when connecting to jolokia in a pod
+      var podName = 'https:' + pod.metadata.name;
```
Comment (liggitt) [inline assets/app/scripts/extensions/javaLink.js]: this should move into `ProxyPod`. When specifying the scheme, the port always has to be provided, even if empty, so you'd need to do `https:name:` for an empty port

Comment (liggitt) [inline assets/app/scripts/extensions/javaLink.js]: or `["https",name,port || ""].join(":")`

**Comment by gashcrumb** [comment]
@liggitt good call, reworked that...

**Comment by liggitt** [comment]
LGTM
https://github.com/openshift/origin/pull/5011 will enable the https:... syntax
Is jolokia ready to serve on TLS?

**Comment by rhuss** [comment]
cool!

yes, since Jolokia 1.3.2 TLS with enhanced client cert check is included. Just pushed a Java STI builder image `fabric8/sti-java-jboss-jdk8` which supports this. The sample project https://github.com/rhuss/sti-camel-cdi can be used for testing (there's also a [create-app.yml](https://github.com/rhuss/sti-camel-cdi/blob/master/create-app.yml))

However, the last missing piece is the RDN against which Jolokia can check the client certificate. Currently it's `cn=OpenShift API Server` but this is of course wrong. Do you have a default name I can insert (can also contain multiples RDNS) ? This can always be overwritten with `AB_JOLOKIA_AUTH_OPENSHIFT` (as well as the CA cert location, which default to `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt`)

Is there a way how I can test it ? I.e. is there a drop in replacement for OpenShift containing this fix available to download ?

**Comment by rhuss** [comment]
@liggitt : Ah,just saw the mail. will add the default RDN to our images.

**Comment by rhuss** [comment]
Good news: I can confirm that an end-to-end test from browser to Jolokia via OS proxy with https works (see screenshot).

What I did:

* Build OpenShift fro @liggitt https_proxy branch in provided Vagrantmachine
* Applied this [build configuration](https://github.com/rhuss/sti-camel-cdi/blob/master/create-app.yml)
* Logged into OpenShift Admin Console
* Took bearer token request Jolokia via Proxy in Browser (see screenshot)

Also checked, that a plain http access didn't work. Next check (tomorrow): Starting the image with a different `clientPrincipal` configured.

image

**Comment by rhuss** [comment]
A negative check when the POD is configured to check agains a different principal:

image

**Comment by liggitt** [comment]
LGTM, [merge]

**Pull request merged**